### PR TITLE
kernel: bump 5.4 to 5.4.176

### PR DESCRIPTION
**Pull request by rsalvaterra**
Manually rebased:
generic/backport-5.4/716-v5.5-net-sfp-move-fwnode-parsing-into-sfp-bus-layer.patch

Signed-off-by: Rui Salvaterra \<rsalvaterra@gmail.com\>

@graysky2, I _think_ I got the rebase right, but I'd love to have more eyeballs on the aforementioned patch.

### CODE REVIEW

File: target/linux/generic/backport-5.4/716-v5.5-net-sfp-move-fwnode-parsing-into-sfp-bus-layer.patch [hunk 1]
```diff
@@ -42,6 +42,10 @@ Signed-off-by: Russell King <rmk+kernel@armlinux.org.uk>
  		return ret;
  	}
  
+@@ -587,9 +580,7 @@ static int phylink_register_sfp(struct p
+ 		return 0;
+ 	}
+
```
Comment (dangowrt) [inline target/linux/generic/backport-5.4/716-v5.5-net-sfp-move-fwnode-parsing-into-sfp-bus-layer.patch:48]: commit https://github.com/gregkh/linux/commit/ff19d70b665d43a90081b7a044e61db051e223cf being part of 5.4.176 made this manual rebase necessary, and it looks good to me.

**Review by dangowrt — APPROVED** [review]

**Comment by Borromini** [comment]
@rsalvaterra I'm testing this on 21.02 HEAD so granted, it's not too kosher. But i've gone through the .171 to .175 bumps and I'm not seeing anything major there. The SFP bit is breaking here:

```
drivers/net/phy/phylink.c: In function 'phylink_register_sfp':
drivers/net/phy/phylink.c:582:34: error: 'ref' undeclared (first use in this function); did you mean 'ret'?
  if (!fwnode_device_is_available(ref.fwnode)) {
                                  ^~~
                                  ret
drivers/net/phy/phylink.c:582:34: note: each undeclared identifier is reported only once for each function it appears in
```
This while trying to compile mt7621. If I compile e.g. ath79/generic, ipq40xx, they compile fine; mt7621 and realtek break though - both with the same error. I'm not sure if MT7621 has any hardware with SFP (I suppose it does), the Realtek SoCs definitely do.

**Comment by Borromini** [comment]
@rsalvaterra To make this less of a wild goose chase, I have applied your PR to master as intended, switched the realtek target manually back to 5.4 and it breaks as well. Ramips has all the 5.4 bits removed on master already, so I cannot test that. But it can definitely be reproduced and is apparently not just affecting master.

```
make[5]: Entering directory '/var/media/data/code/lede/master/build_dir/target-mips_4kec_musl/linux-realtek_generic/linux-5.4.176'
  CALL    scripts/checksyscalls.sh
  CALL    scripts/atomic/check-atomics.sh
  CHK     include/generated/compile.h
  CC      drivers/net/phy/phylink.o
drivers/net/phy/phylink.c: In function 'phylink_register_sfp':
drivers/net/phy/phylink.c:582:41: error: 'ref' undeclared (first use in this function); did you mean 'ret'?
  582 |         if (!fwnode_device_is_available(ref.fwnode)) {
      |                                         ^~~
      |                                         ret
drivers/net/phy/phylink.c:582:41: note: each undeclared identifier is reported only once for each function it appears in
make[8]: *** [scripts/Makefile.build:262: drivers/net/phy/phylink.o] Error 1
make[7]: *** [scripts/Makefile.build:497: drivers/net/phy] Error 2
make[6]: *** [scripts/Makefile.build:497: drivers/net] Error 2
make[5]: *** [Makefile:1734: drivers] Error 2
```

**Comment by rsalvaterra** [comment]
Oh, bugger. Looking into it, thanks for the heads up, @Borromini.

**Comment by rsalvaterra** [comment]
Ok, so [this](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/drivers/net/phy/phylink.c?h=linux-5.4.y&id=ff19d70b665d43a90081b7a044e61db051e223cf) is the change that bit us. Removing that chunk of code obviously fixes the build. I don't know what else might break without it, though. 😕

**Comment by Borromini** [comment]
I have no SFP equipment so I can't test this myself (or comment on the code...).

@bmork Would you have any clue about this?

**Comment by bmork** [comment]
I believe  target/linux/generic/backport-5.4/716-v5.5-net-sfp-move-fwnode-parsing-into-sfp-bus-layer.patch is breaking this

**Comment by bmork** [comment]
Maybe revert the 5.4 LTS backport (ff19d70b665d43a90081b7a044e61db051e223cf) and apply the original commit  2148927e6ed4 ("net: sfp: ignore disabled SFP node") instead?

I haven't tested, but I assume that fits the 5.5 backported code better

**Comment by Borromini** [comment]
@rsalvaterra @bmork Thanks. Updated PR compiles fine on master.

`Build-tested: realtek/generic`

**Comment by graysky2** [comment]
```
Build system: x86_64
Build-tested: ramips/mt7621
```

**Comment by rsalvaterra** [comment]
I'm going to drop this, I completely forgot it even existed and I'm not really looking into it. 😕

**Comment by dangowrt** [comment]
This PR should be rebased on top of the openwrt-21.02 branch now that Linux 5.4 is no longer present in master and openwrt-22.03 branches.

**Pull request closed without merging**